go/enclave/storage/init/edgelessdb/001_init.sql

```diff
@@ -29,24 +29,21 @@ GRANT ALL ON obsdb.attestation_key TO obscuro;
 
 create table if not exists obsdb.block
 (
-    hash         binary(32),
-    parent       binary(32),
+    hash         binary(16),
```

Comment:
Is this safe? Likes does ethereum have hash collisions if you truncate the block hashes to 16bytes and what would happen with this table if there was a collision?

Reply:
Though SHA-256 nominally offers a 256-bit output, no weakness about it is known when the output is truncated to 128 bits, except, of course, weaknesses inherent to the shorter output length; e.g. collision resistance drops from the infeasible 2128 to the possible (but hard) 264.

And it sounds like keccak is similarly secure for truncation. So I guess it's all good?

Reply:
yeah. We'll add checks that the value retrieved from the db hashes to what you expect.
We would have done that anyway as the db must be considered to be vulnerable